api-v2/src/interceptors/transform-response-interceptor.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface ServerResponse<T> {
```

Comment:
now that `ServerResponse` contains no additional information on top of the payload, could we remove it? it's annoying on the frontend to have to do `.data` a lot. axios by default adds a `.data`, so right now our frontend would look like this:

```typescript
const { data, error } = useSWR("/api/v2/students/1", (path) => axios.get(path));
const student = data.data.data;
```

Reply:
ooh yea i can remove it, fetch doesn't add the data section but we should probably be using `axios` anyways lmao